src/scripts.js

```diff
@@ -22,21 +23,85 @@ const toggleSeachOption = document.querySelector(".toggle-search-option");
 const searchButton = document.querySelector(".submit-search");
 const myRecipesButton = document.querySelector(".my-recipes-button");
 const homeButton = document.querySelector(".home-button");
+const myPantryButton = document.querySelector(".my-pantry");
 const keywordList = document.getElementById("keyword-list");
 const keywordSection = document.querySelector(".keyword-section");
 const closeIcon = document.querySelector(".close-icon");
 const saveIcon = document.querySelector(".save-recipe-icon");
+const defaultSearch = document.querySelector(".default-search");
+const pantrySearch = document.querySelector(".pantry-ingredient-datalist");
+const ingredientList = document.querySelector("#ingredient-list");
+const pantryItemTemplate = document.querySelector(".pantry-item-card-template");
+const pantry = document.querySelector(".pantry");
+const datalistString = document.querySelector(".input-value");
+const datalistQty = document.querySelector(".choose-ingredient-amount");
+const datalistButton = document.querySelector(".add-ingredient-quantity");
+const cookButton = document.querySelector(".cook-button");
+const ingredientsNeeded = document.querySelector(".ingredients-needed");
 
+datalistButton.addEventListener("click", updatePantryFromDatalist);
 window.addEventListener("load", loadData);
 toggleSeachOption.addEventListener("click", showKeywords);
 keywordList.addEventListener("click", keywordClicked);
 homeButton.addEventListener("click", displayAllRecipesView);
 searchButton.addEventListener("click", executeSearch);
+myPantryButton.addEventListener("click", showPantry);
+cookButton.addEventListener("click", cookRecipe);
 myRecipesButton.addEventListener("click", displayUserRecipes);
 closeIcon.addEventListener("click", closeSpecificRecipe);
 saveIcon.addEventListener("click", specificRecipeClicked);
 resultCardsContainer.addEventListener("click", specificRecipeClicked);
 
+function updatePantryFromDatalist() {
+  user.evaluatePantry(
+    datalistString.value,
+    parseInt(datalistQty.value),
+    recipeRepo
+  );
+  pantry.innerHTML = "";
+  user.removeFromPantryView();
+  makeIngredientCard();
+}
+
+function cookRecipe() {
+  let recipeToCook = recipeRepo.recipes.find((recipe) => {
+    return recipe.id === parseInt(saveIcon.dataset.id);
+  });
+  user.compareIngredientsNeeded(recipeToCook);
+  if (user.notMatchingIngredients.length >= 1) {
+    user.compareIngredientAmounts(recipeToCook);
```

Comment:
this is the solution! You want to move line 72 outside of that if statement on line 71